src/Xamarin.Android.Build.Tasks/Tasks/ProcessNativeLibraries.cs

```diff
@@ -0,0 +1,66 @@
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Framework;
+
+namespace Xamarin.Android.Tasks
+{
+	/// <summary>
+	/// Processes .so files coming from @(ResolvedFileToPublish).
+	/// * Checks if ABI is valid
+	/// * Fixes up libmonodroid.so based on $(AndroidIncludeDebugSymbols)
+	/// </summary>
+	public class ProcessNativeLibraries : AndroidTask
+	{
+		public override string TaskPrefix => "PRNL";
+
+		/// <summary>
+		/// Assumed to be .so files only
+		/// </summary>
+		public ITaskItem [] InputLibraries { get; set; }
+
+		public bool IncludeDebugSymbols { get; set; }
+
+		[Output]
+		public ITaskItem [] OutputLibraries { get; set; }
+
+		public override bool RunTask ()
+		{
+			if (InputLibraries == null || InputLibraries.Length == 0)
+				return true;
+
+			var output = new List<ITaskItem> (InputLibraries.Length);
+
+			foreach (var library in InputLibraries) {
+				var abi = MonoAndroidHelper.GetNativeLibraryAbi (library);
+				if (string.IsNullOrEmpty (abi)) {
```

Comment:
…especially when I look at *this*: `XA4301_ABI_RID` won't be emitted if the lib *is* in a directory named after an ABI we know about, `%(RuntimeIdentifier)` be damned.

*Presumably* if `%(RuntimeIdentifier)` doesn't "match", it's unusable, e.g. trying to use a "desktop" Linux `.so` on Android is *doomed*, because `libc.so` is in the wrong location (among other things).  Should we "fail early" and error, *and check* before calling `MonoAndroidHelper.GetNativeLibraryAbi()`?  Or should we "fail at runtime" and warn -- as is done in this PR -- if the `.so` can't actually be used?

Which dovetails back with the warning message: if an unsupported `%(RuntimeIdentifier)` means "this has no chance of working", then the message needs to state that, and the fix is to *remove the offending `.so` file*.  If it has *a* chance of working, then `XA4301_ABI_RID` should mirror `XA4301_ABI`, and mention that moving the `.so` into a directory with the ABI name will work.

As-is, the `XA4301_ABI_RID` warning means nothing.

Reply:
I don't believe the `%(RuntimeIdentifier)` metadata would be present in user projects. It would only be present from NuGet packages.

Do you think the existing checks here needs to be changed?

https://github.com/xamarin/xamarin-android/blob/866085a611579ae270289751e46d7388c97f4251/src/Xamarin.Android.Build.Tasks/Utilities/MonoAndroidHelper.cs#L286-L313

I think it will already check `%(RuntimeIdentifier)` before directory names.